### PR TITLE
No filter quotes on PR title.

### PR DESCRIPTION
**Pull request by YuriGizra**
#190
selection_457

### CODE REVIEW

**Comment by YuriGizra** [comment]
@nader77, PR is ready.

File: productivity/modules/custom/productivity_restful/plugins/restful/node/github_prs/1.0/ProductivityGithubPrsResource.class.php [hunk 1]
```diff
@@ -13,6 +13,10 @@ class ProductivityGithubPrsResource extends \ProductivityEntityBaseNode {
   public function publicFieldsInfo() {
     $public_fields = parent::publicFieldsInfo();
 
+    $public_fields['label']['process_callbacks'] = array(
```
Comment (nader77) [inline productivity/modules/custom/productivity_restful/plugins/restful/node/github_prs/1.0/ProductivityGithubPrsResource.class.php]: @YuriGizra It should look like this:
```php
$public_fields['label'] = array(
      'process_callbacks' => array(
        array($this, 'stripQuotes'),
      ),
    );
```

File: productivity/modules/custom/productivity_github/productivity_github.module [hunk 2]
```diff
@@ -218,7 +218,7 @@ function productivity_gituhub_payload_handle_pull_request_opened($payload) {
   }
 
   $node = new stdClass();
-  $node->title = check_plain($payload->pull_request->title);
+  $node->title = htmlspecialchars($payload->pull_request->title, ENT_NOQUOTES, 'UTF-8');
```
Comment (nader77) [inline productivity/modules/custom/productivity_github/productivity_github.module]: Let's save without any checks => `$node->title = $payload->pull_request->title;`
Remove any other check_plain in this code.

File: productivity/modules/custom/productivity_restful/plugins/restful/node/github_prs/1.0/ProductivityGithubPrsResource.class.php [hunk 3]
```diff
@@ -13,6 +13,12 @@ class ProductivityGithubPrsResource extends \ProductivityEntityBaseNode {
   public function publicFieldsInfo() {
     $public_fields = parent::publicFieldsInfo();
 
+    $public_fields['label'] = array (
```
Comment (nader77) [inline productivity/modules/custom/productivity_restful/plugins/restful/node/github_prs/1.0/ProductivityGithubPrsResource.class.php]: Remove this please and let's create a script which goes through all the entities and fixes the titles.
Update script example that runs with drush can be found [here](https://github.com/Gizra/productivity/blob/master/productivity/modules/custom/productivity_time_tracking/scripts/update_time_tracking.php)

**Comment by YuriGizra** [comment]
@nader77, fixed.

**Comment by nader77** [comment]
@YuriGizra You didn't add the script.

**Comment by YuriGizra** [comment]
@nader77 , ready.

**Comment by nader77** [comment]
@YuriGizra Was this tested on existing entities and new ones ? After the code review of course

**Comment by YuriGizra** [comment]
@nader77, yes, It works with existing entities and a new ones.

**Comment by nader77** [comment]
Merged.

**Pull request closed without merging**